packages/core-mobile/app/new/routes/(signedIn)/(tabs)/portfolio/index.tsx

```diff
@@ -174,17 +200,42 @@ const PortfolioHomeScreen = (): JSX.Element => {
         <ActionButtons buttons={ACTION_BUTTONS} />
       </View>
     )
-  }
+  }, [
+    ACTION_BUTTONS,
+    animatedHeaderStyle,
+    formattedBalance,
+    isLoading,
+    activeAccount?.name,
+    selectedCurrency,
+    theme.colors.$surfacePrimary,
+    totalPriceChanged,
+    indicatorStatus,
+    formattedPercent,
+    balanceAccurate,
+    handleBalanceHeaderLayout
+  ])
+
+  const handleSelectSegment = useCallback(
+    (index: number): void => {
+      selectedSegmentIndex.value = index
 
-  const handleSelectSegment = (index: number): void => {
-    if (tabViewRef.current?.getCurrentIndex() !== index) {
-      tabViewRef.current?.setIndex(index)
-    }
-  }
+      InteractionManager.runAfterInteractions(() => {
+        if (tabViewRef.current?.getCurrentIndex() !== index) {
+          tabViewRef.current?.setIndex(index)
+        }
+      })
+    },
+    [selectedSegmentIndex]
```

Comment:
@onghwan I adjusted how CollapsibleTabs.Container and SegmentedControl work together 

**before:**
on segment clicked -> we tell pager view to change index -> when pager view broadcasts the index change event, we set segmentIndex value.
this delays the visual change on segment and also makes it kinda janky since a lot of things are happening in between js thread, native thread and UI thread.

**after:**
on segment clicked -> we set segmentIndex value so the segment is clicked visually right away -> we tell pager view to change index only when app is not too busy (initial app unlock, app is busy making a lot of requests).